### ISSUE TITLE
Cannot export a component's prototype function to the references scope

### ISSUE THREAD

**Issue by dylanrtt** [canjs/canjs#2051]
I should be able to export a function from a component like this:

```html
<foo-bar {^@foo}="*bar"></foo-bar>
{{*bar()}}
```

It *should* work for functions defined on the component's viewmodel or the viewmodel's prototype, but neither seems to work.

http://jsbin.com/zumebacuma/edit?html,js,console,output

**Comment by dylanrtt** [comment]
I gave this a try on master and the `foo` function is running 3 times, or just twice if `{{*bar()}}` is removed. The whole point is that it shouldn't be running at all with `@`.

http://jsbin.com/pezologeyi/edit?html,js,console,output

I also tried using the function on the viewmodel instance instead of the prototype method and that doesn't work (never runs). I know it's not mentioned in title of this issue but I think that should work too if it's not too much trouble.

**Comment by justinbmeyer** [comment]
arg.  will look at it.

**Comment by justinbmeyer** [comment]
The problem here seems to be that a compute is built around `*refKey`:

```
{^@method}='*refKey' 
```

So, when the reference scope's `*refKey` vale is set, the compute tries to update its internal value.  So, it tries to read `*refKey` instead of `@*refKey`.

I'm not sure what the best solution is.  One possibility is to get people to write (and make this work):

```
<foo-bar {^@foo}="@*bar"></foo-bar>
```

I think this makes sense because this is what would have to be written if two-way binding a function was ever to be done.  Ugly, but works.

Alternatively, I could special case one way bindings to not create a compute around the value that is going to be set.

I might need to make both work.

**Comment by justinbmeyer** [comment]
Ok, so I made this work right now the ugly way: `<foo-bar {^@foo}="@*bar"></foo-bar>`.  I'm not happy with this.  But it will at least let you move forward and us get out a 2.3.2.

To make it work without the double `@`, it seems to require a bit of additional code.  

Currently, we know which context to set because we always read the current value and set the first context that has a value.  If no context has a value we set the top of the scope.

To set a value, we'd need some similar logic, but something that is able to tell if a property is in some context of the scope without actually reading it.  

Essentially, I would need some type of "scope.hasOwnProperty". But this would extend to `can.compute.read` and eventually to maps.

This is a problem I've know about, but it wasn't a problem because computes were always read first, so the set context was known.  Always seemed like a cheat.
